tests/ui/print_literal.rs

```diff
@@ -42,4 +42,7 @@ fn main() {
     // The string literal from `file!()` has a callsite span that isn't marked as coming from an
     // expansion
     println!("file: {}", file!());
+
+    // Braces in unicode escapes should not be escaped
+    println!("{}", "{} \u{ab123} {}");
```

Comment:
How about a test case with a format string of `"\\u{ab123}"` with binding of `ab123` to something?

Or a test case with additional format arg things like debug output or width specification?

Just to make sure we cover all cases.

Reply:
`\\u{..}` was a good call, it didn't handle that correctly so I've replaced the regex with a state machine